staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -162,144 +162,174 @@ func ValidateCustomResourceDefinitionNames(names *apiextensions.CustomResourceDe
 	return allErrs
 }
 
+// additionalSpecValidator applies validations for different OpenAPI specfication versions.
+type additionalSpecValidator interface {
+	validateCustomResourceDefinitionOpenAPISpec(spec *apiextensions.JSONSchemaProps, fldPath *field.Path) field.ErrorList
```

Comment:
I would simply call this function `validate`. The interface (and the type name) are descriptive enough about what it is validating.